src/main/java/co/omise/Client.java

```diff
@@ -433,7 +419,7 @@ public SourceResource sources() {
      * @throws IOException    the general I/O error that could happen during deserialization
      * @throws OmiseException the custom exception thrown for response errors
      */
-    public <T extends Model, R extends Request<T>> T sendRequest(R request, TypeReference<T> typeReference) throws IOException, OmiseException {
+    public <T extends OmiseList, S extends Model, R extends Request<S>> T sendRequest(R request, TypeReference<T> typeReference) throws IOException, OmiseException {
```

Comment:
@Namiii not sure i understand this signature fully. what's the relation between `T`, `S` and `Request<S>`?

Reply:
@chakrit T is just the return type, which is a ScopedList (child of OmiseList), S is the Model that is used to make the request, ie Request<Charge>.

Reply:
@Namiii but the type parameter of the request class specifies what should be returned right?

Can we do this? 👇

```java
public <T extends OmiseList, R extends Request<T>> T sendRequest(
  R request,
  TypeReference<T> typeReference
)
```

Reply:
What I mean is, they should be a relation between `T` and `S` because it's used together, right?

Reply:
@chakrit That's how it originally was but that didn't work. `Request<T extends Model>` prevented us from supplying it a `T extends OmistList`. We either had to do this or create a Requester that took in a child of OmiseList. OR is there something else I'm missing here?

Reply:
`OmiseList` should extends from `Model` (or some other, more generic base) 😂 

I think that's the root cause. I'm not sure how it was done originally but we might want to think about that.

Reply:
Or maybe we need interfaces here as opposed to concrete class, not sure 😂

Reply:
I think they both extend from the very same base class (`OmiseList` and `Model` I mean). Let me see if I can make some changes that way.